fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    /**
+     * Use fetch to send a GET request to `/api/todos`.
+     * Then 😉, once the response JSON is received and parsed,
+     * update state with the received todos.
+     */
+    fetch('http://localhost:3000/api/todos')
+      .then(res => res.json())
+      .then(data => this.setState({ todos: data }));
+  }
+
+  addTodo(newTodo) {
+    /**
+    * Use fetch to send a POST request to `/api/todos`.
+    * Then 😉, once the response JSON is received ansd parsed,
+    * add the created todo to the state array.
+    *
+    * Do not mutate the original state array, nor any objects within it.
+    * https://reactjs.org/docs/optimizing-performance.html#the-power-of-not-mutating-data
+    *
+    * TIP: Be sure to SERIALIZE the todo object in the body with JSON.stringify()
+    * and specify the "Content-Type" header as "application/json"
+    *
+    * TIP: Use Array.prototype.concat to create a new array containing the contents
+    * of the old array, plus the object returned by the server.
+    */
+
+    fetch('http://localhost:3000/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    })
+      .then(res => res.json());
+
+  }
+
+  toggleCompleted(todoId) {
+
+    const newCompleted = this.state.todos[todoId - 1];
+    let newIsComplete = {};
+    if (newCompleted.isCompleted) {
+      newCompleted.isCompleted = false;
+      newIsComplete = { isCompleted: false };
+    } else if (!newCompleted.isCompleted) {
+      newCompleted.isCompleted = true;
+      newIsComplete = { isCompleted: true };
+    }
+
+    fetch('http://localhost:3000/api/todos/' + todoId, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newIsComplete)
+    })
+      .then(res => res.json())
+      .then(data => {
+        this.setState(newCompleted);
```

Comment:
You'll need to replace the existing, outdated todo with the one you received back from the server (your `data` parameter on line 74).